src/Controller/TaskController.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Controller;
+
+use App\Dto\TaskDto;
+use App\Entity\TaskCalendar;
+use Doctrine\ORM\EntityManagerInterface;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\ParamConverter;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Routing\Annotation\Route;
+use Symfony\Component\Validator\ConstraintViolationListInterface;
+
+class TaskController extends AbstractController
+{
+    #[Route('/tasks', name: 'update_task', methods: ['PATCH'])]
+    #[ParamConverter('task', converter: 'fos_rest.request_body')]
+    public function update(
+        TaskDto $task,
+        ConstraintViolationListInterface $validationErrors,
+        EntityManagerInterface $entity_manager
+    ): JsonResponse {
+        if ($validationErrors->count() > 0) {
+            return $this->json($validationErrors, Response::HTTP_BAD_REQUEST);
+        }
+
+        // TODO this logic should extracted to service probably, but this is a simple crud
+        $task_calendar = $entity_manager->getRepository(TaskCalendar::class)->find($task->id);
```

Comment:
Better will be here to inject the TaskCalendarRepository

Reply:
Choosing that approach will help you during writing tests, because when you will need to mock that part, you will need to create only one mock instead two for the EntityManager and TaskCalendarRepository